uhabits-core/src/jvmTest/java/org/isoron/uhabits/core/ui/NotificationTrayTest.kt

```diff
@@ -0,0 +1,125 @@
+    }
+
+    private var preferences = MockPreferences()
+    private lateinit var notificationTray: NotificationTray
+
+    class DummyStorage : Storage {
+        override fun clear() {
+            throw NotImplementedError("Mock implementation missing")
+        }
+
+        override fun getBoolean(key: String, defValue: Boolean): Boolean {
+            throw NotImplementedError("Mock implementation missing")
+        }
+
+        override fun getInt(key: String, defValue: Int): Int {
+            throw NotImplementedError("Mock implementation missing")
+        }
+
+        override fun getLong(key: String, defValue: Long): Long {
+            throw NotImplementedError("Mock implementation missing")
+        }
+
+        override fun getString(key: String, defValue: String): String {
+            throw NotImplementedError("Mock implementation missing")
+        }
+
+        override fun onAttached(preferences: Preferences) {
+        }
+
+        override fun putBoolean(key: String, value: Boolean) {
+            throw NotImplementedError("Mock implementation missing")
+        }
+
+        override fun putInt(key: String, value: Int) {
+            throw NotImplementedError("Mock implementation missing")
+        }
+
+        override fun putLong(key: String, value: Long) {
+            throw NotImplementedError("Mock implementation missing")
+        }
+
+        override fun putString(key: String, value: String) {
+            throw NotImplementedError("Mock implementation missing")
+        }
+
+        override fun remove(key: String) {
+            throw NotImplementedError("Mock implementation missing")
+        }
+    }
+
+    class MockPreferences : Preferences(DummyStorage()) {
+        private var activeNotifications: HashMap<Habit, NotificationTray.NotificationData> =
+            HashMap()
+
+        override fun setActiveNotifications(activeNotifications: Map<Habit, NotificationTray.NotificationData>) {
+            this.activeNotifications = HashMap(activeNotifications)
+        }
+
+        override fun getActiveNotifications(habitList: HabitList): HashMap<Habit, NotificationTray.NotificationData> {
+            return activeNotifications
+        }
+    }
+
+    @Before
+    @Throws(Exception::class)
+    override fun setUp() {
+        super.setUp()
+        notificationTray =
+            NotificationTray(taskRunner, commandRunner, preferences, systemTray, habitList)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun testShow() {
+        // Show a reminder for a habit
+        val habit = fixtures.createEmptyHabit()
+        habit.reminder = Reminder(8, 30, WeekdayList.EVERY_DAY)
+        val timestamp = Timestamp(System.currentTimeMillis())
+        val reminderTime = System.currentTimeMillis()
+        notificationTray.show(habit, timestamp, reminderTime)
+
+        // Verify that the active notifications include exactly the one shown reminder
+        // TODO are we guaranteed that task has executed?
```

Comment:
As a `ShowNotificationTask` is executed by a task runner, I am not sure whether this might be asynchronous and we should for example add a delay here to increase the likelihood that it has executed? Or are tasks (at least for these tests) run in the same thread, which would make tests safer?